pkg/handler/defi_test.go

```diff
@@ -146,3 +150,317 @@ func TestHandler_GetCoin(t *testing.T) {
+			w := httptest.NewRecorder()
+			ctx, _ := gin.CreateTestContext(w)
+			ctx.Request = httptest.NewRequest("GET", fmt.Sprintf("/api/v1/defi/watchlist?user_id=%s", tt.userId), nil)
+			coingeckoMock.EXPECT().GetCoinsMarketData(tt.defaultTickers).Return(tt.coinGeckoDefaultTickersResponse, nil, 200).AnyTimes()
+
+			h.GetUserWatchlist(ctx)
+			require.Equal(t, tt.wantCode, w.Code)
+			expRespRaw, err := ioutil.ReadFile(tt.wantResponsePath)
+			require.NoError(t, err)
+			require.JSONEq(t, string(expRespRaw), w.Body.String(), "[Handler.GetUserWatchlist] response mismatched")
+		})
+	}
+}
+
+func TestHandler_AddToWatchlist(t *testing.T) {
+	cfg := config.LoadTestConfig()
+	db := testhelper.LoadTestDB("../../migrations/test_seed")
+	repo := pg.NewRepo(db)
+	log := logger.NewLogrusLogger()
+	s := pg.NewPostgresStore(&cfg)
+	svc, _ := service.NewService(cfg, log)
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	coingeckoMock := mock_coingecko.NewMockService(ctrl)
+	//mock_coingeckosupportedtokens := mock_coingeckosupportedtokens.NewMockStore(ctrl)
+	//repo.CoingeckoSupportedTokens = mock_coingeckosupportedtokens
+	svc.CoinGecko = coingeckoMock
+	entityMock := entities.New(cfg, log, repo, s, nil, nil, nil, svc, nil, nil, nil)
+	h := &Handler{
+		entities: entityMock,
+		log:      log,
+	}
+	tests := []struct {
+		name                     string
+		req                      request.AddToWatchlistRequest
+		res                      *response.AddToWatchlistResponse
+		coingeckoSupportedTokens []model.CoingeckoSupportedTokens
+		wantError                error
+		wantCode                 int
+		wantResponsePath         string
+	}{
+		// TODO: Add test cases.
+		{
+			name: "success - found suggestions",
+			req: request.AddToWatchlistRequest{
+				UserID: "319132138849173505",
+				Symbol: "doge",
+			},
+			res: &response.AddToWatchlistResponse{
+				Data: &response.AddToWatchlistResponseData{
+					BaseSuggestions: []model.CoingeckoSupportedTokens{
+						{
+							ID:     "binance-peg-dogecoin",
+							Symbol: "doge",
+							Name:   "Binance-Peg Dogecoin",
+						},
+						{
+							ID:     "dogecoin",
+							Symbol: "doge",
+							Name:   "Dogecoin",
+						},
+					},
+					TargetSuggestions: nil,
+				},
+			},
+			coingeckoSupportedTokens: []model.CoingeckoSupportedTokens{
+				{
+					ID:     "binance-peg-dogecoin",
+					Symbol: "doge",
+					Name:   "Binance-Peg Dogecoin",
+				},
+				{
+					ID:     "dogecoin",
+					Symbol: "doge",
+					Name:   "Dogecoin",
+				},
+			},
+			wantCode:         200,
+			wantResponsePath: "testdata/user_watchlist/post-200-ok.json",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			body, err := json.Marshal(tt.req)
+			if err != nil {
+				t.Error(err)
+				return
+			}
+
+			w := httptest.NewRecorder()
+			ctx, _ := gin.CreateTestContext(w)
+			ctx.Request = httptest.NewRequest("POST", "/api/v1/defi/watchlist", bytes.NewBuffer(body))
```

Comment:
add header `application/json` or change binding type to `BindJSON` or something like that